src/firefly/js/visualize/Histogram.jsx

```diff
@@ -27,16 +29,55 @@ var Histogram = React.createClass(
 
         getDefaultProps() {
             return {
-                data: undefined,
-                source: undefined,
-                height: 300,
-                logs: undefined,
-                reversed: undefined,
                 desc: 'Sample Distribution',
                 binColor: '#d1d1d1'
             };
         },
 
+        shouldComponentUpdate(nextProps) {
```

Comment:
What I am trying to do here is to update the existing chart rather then
re-render:  true is returned by a single line in the beginning. But I have
added a comment.

2016-05-26 10:30 GMT-07:00 loitly <notifications@github.com>:

> In src/firefly/js/visualize/Histogram.jsx
> <https://github.com/Caltech-IPAC/firefly/pull/92#discussion_r64786354>:
>
> >                  desc: 'Sample Distribution',
> >                  binColor: '#d1d1d1'
> >              };
> >          },
> >
> > +        shouldComponentUpdate(nextProps) {
>
> this is purely my opinion without studying the code.
> there is a lot of logic in this check. it would be helpful if there is a
> short comments outlining when this component should update.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/Caltech-IPAC/firefly/pull/92/files/8f6603b9fd61980983a353c464e70a6153ece3a0#r64786354>
>